### PR TITLE
feat(cb2-10672): adr certificate generation is available when a pass result is selected for adr contingency tests

### PR DESCRIPTION
**Pull request by owen-corrigan-bjss**
## Ticket title

contingency test flow for generating ADR certificates. 

_One line description_
[CB2-XXXX](https://dvsa.atlassian.net/browse/CB2-XXXX)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      <app-field-error-message for="view-button" [error]="errorString"></app-field-error-message>
+      <app-button
+        id="view-button"
+        [params]="documentParams(fileName)"
+        [fileName]="fileName"
+        appRetrieveDocument
+        [loading]="true"
+        type="link"
+        design="link"
+        >view certificate</app-button
```
Comment (shivangidas) [inline src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html:24]:
```suggestion
        >View certificate</app-button
```

**Review by shivangidas — DISMISSED** [review]

File: src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<ng-container>
```
Comment (pbardy2000) [inline src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html:1]: Do you need an empty ng-container, or is this missing an *ngIf?

File: src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      ? `An ADR certificate was last generated on ${new Date(sortedTests[0].generatedTimestamp).toLocaleDateString('en-UK')}`
+      : 'There are no previous ADR certificates for this vehicle';
+  }
+
+  documentParams(certificate: string): Map<string, string> {
+    return new Map([['fileName', certificate]]);
+  }
+
+  handleSubmit(): void {
+    if (this.store) {
```
Comment (pbardy2000) [inline src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.ts:76]: This seems unnecessary, is there a reason why the store would be undefined at this point? You're injecting it via DI.

**Review by tomcrawleyy — APPROVED** [review]

File: src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html (hunk 2 above)
Comment (owen-corrigan-bjss) [reply src/app/forms/components/contingency-adr-generate-cert/contingency-adr-generate-cert.component.html]: Think I must have tried something with it and left it in place. I'll remove it

**Review by pbardy2000 — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Pull request merged**